### PR TITLE
Only print parameters with elided lifetimes in elision error messages.

### LINKED ISSUE #30255: "0 elided lifetimes" is still confusing
```rust
struct S<'a> {
    field: &'a i32,
}
fn f(a: &S, b: i32) -> &i32 {
    panic!();
}
```

```
test.rs:4:24: 4:28 help: this function's return type contains a borrowed value,
but the signature does not say whether it is borrowed from one of `a`'s 2 elided lifetimes
or one of `b`'s 0 elided lifetimes
```

See #30086 for a similar issue which has been fixed.

### PR DESCRIPTION
**Pull request by nham**
When displaying the function parameters for a lifetime elision error message,
this changes it to first filter out the parameters that don't have elided
lifetimes.

Fixes #30255.

### CODE REVIEW

File: src/librustc_typeck/astconv.rs [hunk 1]
```diff
@@ -215,15 +215,24 @@ fn report_elision_failure(
...
+                                       .filter(|info| info.lifetime_count > 0)
+                                       .collect();
+
+    let elided_len = elided_params.len();
+
+    let any_lifetimes = if elided_len > 0 {
+        true
+    } else {
+        false
+    };
```
Comment (sanxiyn) [inline src/librustc_typeck/astconv.rs]: This is just `let any_lifetimes = elided_len > 0`.

Comment (sanxiyn) [inline src/librustc_typeck/astconv.rs]: Better, replace `!any_lifetimes` below with `elided_len == 0`.

Comment (nham) [inline src/librustc_typeck/astconv.rs]: Good point. I've done this.

**Pull request merged**